### PR TITLE
Added a way to override default values in config for docs

### PR DESCRIPTION
**Pull request by spacecowboy**
Some settings are calculated at runtime and depend on the current
machine. This is problematic for documentation which uses the default
values and prints them in the manual. This change adds an Annotation
which can specify what to  put in the manual if the actual default
value is not suitable.

Also added the necessary plumbing to access the information in the
annotations via the Config class.

### CODE REVIEW

**Review by praveenag — APPROVED** [review]

File: community/configuration/src/main/java/org/neo4j/configuration/ConfigOptions.java [hunk 1]
```diff
@@ -24,7 +24,9 @@
 import java.util.Optional;
 import java.util.stream.Collectors;
 import javax.annotation.Nonnull;
+import javax.swing.text.html.Option;
```
Comment (jjaderberg) [inline community/configuration/src/main/java/org/neo4j/configuration/ConfigOptions.java]: Over-zealous IDE? Both new imports are unused.

File: community/kernel/src/main/java/org/neo4j/kernel/configuration/ConnectorValidator.java [hunk 2]
```diff
@@ -20,6 +20,7 @@
 package org.neo4j.kernel.configuration;
 
 import java.util.Arrays;
+import java.util.Collections;
```
Comment (jjaderberg) [inline community/kernel/src/main/java/org/neo4j/kernel/configuration/ConnectorValidator.java]: unused

**Review by jjaderberg — CHANGES_REQUESTED** [review]
This is a nice way to handle defaults that can't be documented. Can we expose whether a setting is deprecated and what its valid values restrictions are in a similar way?

**Comment by spacecowboy** [comment]
@jjaderberg corrected the unused imports. I will do the deprecated work separately

**Review by jjaderberg — APPROVED** [review]

**Pull request merged**